src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchOptions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    internal class FindReferencesSearchOptions
+    {
+        public static readonly FindReferencesSearchOptions Default = 
+            new FindReferencesSearchOptions(associatePropertyReferencesWithSpecificAccessor: false);
+
+        /// <summary>
+        /// When searching for property, associate specific references we find to the relevant
+        /// accessor symbol (if there is one).  For example, in C#, this would result in:
+        /// 
+        ///     P = 0;     // A reference to the P.set accessor
+        ///     var v = P; // A reference to the P.get accessor
+        ///     P++;       // A reference to P.get and P.set accessors
+        ///     nameof(P); // A reference only to P.  Not associated with a particular accessor.
+        ///     
+        /// The default for this is false.  With that default, all of the above references
+        /// are associated with the property P and not the accessors.
+        /// </summary>
+        public bool AssociatePropertyReferencesWithSpecificAccessor { get; }
+
+        public FindReferencesSearchOptions(
+            bool associatePropertyReferencesWithSpecificAccessor)
+        {
+            AssociatePropertyReferencesWithSpecificAccessor = associatePropertyReferencesWithSpecificAccessor;
+        }
+
+        public FindReferencesSearchOptions WithAssociatePropertyReferencesWithSpecificAccessor(
+            bool associatePropertyReferencesWithSpecificAccessor)
+        {
+            return new FindReferencesSearchOptions(associatePropertyReferencesWithSpecificAccessor);
+        }
+
+        /// <summary>
+        /// For IDE features, if the user starts searching on an accessor, then we want to give
+        /// results associated with the specific accessor.  Otherwise, if they search on a property,
+        /// then associate everything with the property.
+        /// </summary>
+        public static FindReferencesSearchOptions GetFeatureOptionsForStartingSymbol(ISymbol symbol)
```

Comment:
>GetFeatureOptionsForStartingSymbol [](start = 50, length = 34)

nit: the "Features" in this method name seems odd. Maybe `GetReferencesSearchOptions(ISymbol startingSymbol)`?

Reply:
So, htis is specifically about getting the options when running the "find refs feature". i.e. the experience you get if you "shift f12".  This is a distinct concept of "what do you get if you run find-refs, 'the api'".  The latter obviously doesn't care (since you provide the options when you call the API).  So this only about having the options be right if you use "the feature".  let me know if you need more info.